src/scrabble.js

```diff
@@ -1,6 +1,98 @@
 class Scrabble {
-  // Write your implementation here
+  constructor (word) {
+    this.word = word
+  }
 
+  scoreLoop (string) {
+    let num = 0
+    for (let i = 0; i < string.length; i++) {
+      if (string[i].match(/[^A-Z{}[]]/i)) {
+        return 0
+      }
+      if (string[i].match(/[AEIOULNRST]/i)) {
+        num++
+      }
+      if (string[i].match(/[DG]/i)) {
+        num += 2
+      }
+      if (string[i].match(/[BCMP]/i)) {
+        num += 3
+      }
+      if (string[i].match(/[FHVWY]/i)) {
+        num += 4
+      }
+      if (string[i].match(/[K]/i)) {
+        num += 5
+      }
+      if (string[i].match(/[JX]/i)) {
+        num += 8
+      }
+      if (string[i].match(/[QZ]/i)) {
+        num += 10
+      }
+    }
+    return num
+  }
+
+  scoreMain () {
+    if (this.word == null) {
+      return 0
+    }
+
+    const wordTrimmed = this.word.trim()
+
+    return this.scoreLoop(wordTrimmed)
+  }
+
+  scoreDoubleTripleLetter (open, close) {
```

Comment:
This is excellent! It's great to see you using the arguments to prevent yourself from having duplicated code!